src/basic/cgroup-util.c

```diff
@@ -1295,26 +1295,32 @@ static const char *skip_user_manager(const char *p) {
         p += strspn(p, "/");
 
         n = strcspn(p, "/");
-        if (n < STRLEN("user@x.service"))
+        if (n < STRLEN("user@x.service")) /* || n < STRLEN("capsule@x.service")) */
                 return NULL;
 
-        if (memcmp(p, "user@", 5) == 0 && memcmp(p + n - 8, ".service", 8) == 0) {
-                char buf[n - 5 - 8 + 1];
+        _cleanup_free_ char *i = NULL;
+        const char *unit_name = strndupa_safe(p, n);
```

Comment:
maybe don't use strndupa() here anymore. the input string might be attacker controlled, who knows... just use strndup(), htere's no shame in that

Reply:
or if you want to avoid the oom handling, then please add a check that validates that `n` is <= NAME_MAX before you do the strndupa() her

Reply:
OK, I'll do dynamic allocation, user units may be deep hierarchies.